docs/usage/basics.ipynb

```diff
@@ -0,0 +1,1040 @@
+     "hide-input"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "x_domain = np.linspace(0, 5, num=200)\n",
+    "y_values = function_1d({\"x\": x_domain})\n",
+    "fig = plt.figure(figsize=(8, 5))\n",
+    "plt.plot(x_domain, y_values)\n",
+    "plt.gca().set_xlabel(\"$x$\")\n",
+    "plt.gca().set_ylabel(\"$f(x)$\")\n",
+    "\n",
+    "x_min = x_range[1]\n",
+    "x_max = x_range[2]\n",
+    "y_max = 0.21\n",
+    "x_value = 1.5\n",
+    "\n",
+    "line_segment = [[0, 0], [0, y_max]]\n",
+    "plt.plot(*line_segment, color=\"black\")\n",
+    "plt.text(\n",
+    "    -0.22,\n",
+    "    y_max / 2 * 0.5,\n",
+    "    \"uniform sample $(0, y_{max})$\",\n",
+    "    rotation=\"vertical\",\n",
+    ")\n",
+    "plt.axhline(y=y_max, linestyle=\"dotted\", color=\"black\")\n",
+    "plt.text(\n",
+    "    x_min + 0.1,\n",
+    "    y_max - 0.01,\n",
+    "    \"$y_{max}$\",\n",
+    ")\n",
+    "\n",
+    "line_segment = [[x_min, x_max], [0, 0]]\n",
+    "plt.plot(*line_segment, color=\"black\")\n",
+    "plt.text(\n",
+    "    (x_max - x_min) / 2 - 0.22,\n",
+    "    0.005,\n",
+    "    R\"uniform sample $(x_\\mathrm{min}, x_\\mathrm{max})$\",\n",
+    ")\n",
+    "plt.scatter(x_value, function_1d({\"x\": x_value}))\n",
+    "plt.axvline(x=x_value, linestyle=\"dotted\")\n",
+    "\n",
+    "\n",
+    "def draw_y_hit(x_random, y_random):\n",
+    "    y_value = function_1d({\"x\": x_random})\n",
+    "    color = \"green\" if y_random < y_value else \"red\"\n",
+    "    text = \"hit\" if y_random < y_value else \"miss\"\n",
+    "    plt.scatter(0, y_random, color=color)\n",
+    "    plt.arrow(\n",
+    "        x=0,\n",
+    "        y=y_random,\n",
+    "        dx=x_random,\n",
+    "        dy=0,\n",
+    "        head_length=0.15,\n",
+    "        length_includes_head=True,\n",
+    "        color=color,\n",
+    "        linestyle=\"dotted\",\n",
+    "    )\n",
+    "    plt.text(x_value + 0.05, y_random, text)\n",
+    "\n",
+    "\n",
+    "draw_y_hit(x_random=x_value, y_random=0.05)\n",
+    "draw_y_hit(x_random=x_value, y_random=0.17)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "There is one problem though: _how to determine $y_\\mathrm{max}$?_ In this example, we can just read off the value of $y_\\mathrm{max}$, or even compute it analytically from the original {class}`sympy.Expr <sympy.core.expr.Expr>`. This is not the case generally though, so we need to apply a trick.\n",
+    "\n",
+    "Since we are generating **uniformly distributed** random values values of $x$ and computing their $f(x)$ values, we can keep track of which values of $f(x)$ is the highest. Starting with $y_\\mathrm{max} = 0$ we just set $y_\\mathrm{max} = f(x)$ once $f(x) > y_\\mathrm{max}$ and _completely restart_ the generate loop. Eventually, some value of $x$ will lie near the absolute maximum of $f$ and the data generation will happily continue until the requested number of events has been reached."
```

Comment:
It should be noted that this procedure only works if the number of generated events is large. If the number of generated events is small the distribution generated by the this procedure will not accurately reflect the original function.

Reply:
Added a warning admonition